### PR TITLE
Ticket 4: 5-get-articles-comments:Implememtation path GET /api/articles/:article_id/comments

### PR DESCRIPTION
**Pull request by AstridKMina**
This Pull Request implements the functionality to retrieve comments for a specific article via the GET /api/articles/:article_id/comments endpoint.

Key Changes:
Added the fetchArticleCommentsById function to query the database for comments associated with a given article, sorted in descending order by creation date.
Implemented the getArticleCommentsById controller to handle requests, validate article_id, and manage errors for invalid input or non-existent articles.
Added test cases for GET /api/articles/:article_id/comments to verify:
A successful response returning a list of comments.
Handling of articles with no comments.
A 400 error response for an invalid article_id.
A 404 error response for a non-existent article_id.
Updated API documentation with a description of the endpoint and an example response.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -174,5 +174,61 @@ describe("/api/articles?order=desc", () => {
...
+    comments.forEach((comment) => {
+      expect(comment).toHaveProperty("author")
+      expect(comment).toHaveProperty("comment_id")
+      expect(comment).toHaveProperty("article_id")
+      expect(comment).toHaveProperty("body")
+      expect(comment).toHaveProperty("created_at")
+      expect(comment).toHaveProperty("votes")
+    })
+  })
+  test("200: Responds with an array of comments objects", async () => {
```
Comment (nc-bela-bertalan) [inline __tests__/app.test.js:198]: This test is unnecessary as you used forEach in the previous test which only works on arrays, so if 'comments' wasn't an array, that test would fail.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -174,5 +174,61 @@ describe("/api/articles?order=desc", () => {
...
+  test("404: Responds with an error message for not found article_id", async () => {
+
+    const response = await request(app)
+      .get("/api/articles/999/comments")
+      .expect(404);
+
+    const error = response.body;
+
+    expect(error).toEqual({ msg: "Article not found" });
+  })
```
Comment (nc-bela-bertalan) [inline __tests__/app.test.js:231]: Please add another test there you expect 200 and an empty array of comments where there are no comments for an article yet. 

You will need to check the reason for the query returning empty rows:
article exists but has no comments.
article does not exist.

**Review by nc-bela-bertalan — COMMENTED** [review]
Hi Astrid, nice work but please see my other comments too!
